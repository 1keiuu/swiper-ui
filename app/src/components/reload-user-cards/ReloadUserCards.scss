.reload-user__container {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  p {
    font-weight: bold;
    text-align: center;
    margin: 0;
    line-height: 40px;
    font-size: 17px;
  }
}

.empty-user__container {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  img {
    width: 70%;
  }
  p {
    font-weight: bold;
    text-align: center;
    margin: 0;
    line-height: 40px;
    font-size: 17px;
  }
}
