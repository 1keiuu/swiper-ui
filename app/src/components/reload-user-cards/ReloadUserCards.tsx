import React, { useContext, useEffect, useState } from "react";
import "./ReloadUserCards.scss";
import { CurrentCardStore } from "../../store/CurrentCard";
import fetch from "../../lib/fetch";
import { UserStore } from "../../store/User";

const ReloadUserCards: React.FC = () => {
  const { state, dispatch } = useContext(UserStore);
  const [status, setStatus] = useState("reloading");

  useEffect(() => {
    fetch((res) => {
      if (res.length == 0) setStatus("empty");
      dispatch({ type: "APPEND_USERS", incomingUsers: res });
    });
  }, []);
  if (status == "reloading") return <div>取得中...</div>;
  else if (status == "empty") return <div>empty</div>;
};

export default ReloadUserCards;
