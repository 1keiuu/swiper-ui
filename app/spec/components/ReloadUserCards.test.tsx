import React from "react";
import { cleanup, render } from "@testing-library/react";
import ReloadUserCards from "../../src/components/reload-user-cards/ReloadUserCards";

describe("ReloadUserCards", () => {
  test("render as before.", () => {
    const { asFragment } = render(<ReloadUserCards />);
    expect(asFragment()).toMatchSnapshot();
  });
});
