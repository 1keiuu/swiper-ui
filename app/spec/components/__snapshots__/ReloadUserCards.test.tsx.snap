// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReloadUserCards render as before. 1`] = `
<DocumentFragment>
  <div
    class="reload-user__container"
  >
    <p>
      取得中...
    </p>
  </div>
</DocumentFragment>
`;
